src/SignalR/common/Protocols.Json/src/Protocol/JsonHubProtocol.cs

```diff
@@ -746,19 +735,20 @@ private HubMessage ApplyHeaders(HubMessage message, Dictionary<string, string> h
 
         internal static JsonSerializerOptions CreateDefaultSerializerSettings()
         {
-            var options = new JsonSerializerOptions();
-            options.WriteIndented = false;
-            options.ReadCommentHandling = JsonCommentHandling.Disallow;
-            options.AllowTrailingCommas = false;
-            options.IgnoreNullValues = false;
-            options.IgnoreReadOnlyProperties = false;
-            options.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
-            options.PropertyNameCaseInsensitive = true;
-            options.MaxDepth = 64;
-            options.DictionaryKeyPolicy = null;
-            options.DefaultBufferSize = 16 * 1024;
-
-            return options;
+            return new JsonSerializerOptions()
+            {
+                WriteIndented = false,
+                ReadCommentHandling = JsonCommentHandling.Disallow,
+                AllowTrailingCommas = false,
+                IgnoreNullValues = false,
+                IgnoreReadOnlyProperties = false,
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                PropertyNameCaseInsensitive = true,
+                MaxDepth = 64,
+                DictionaryKeyPolicy = null,
+                DefaultBufferSize = 16 * 1024,
+                Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
```

Comment:
The "Unsafe" in the encoder name makes me nervous.

Looking at https://github.com/dotnet/corefx/issues/39410 it seems the encoder defaulted to using escape sequences for a bunch of characters so the strings could can be safely used inside a HTML element or attribute without an additional sanitation step. Since SignalR no longer has a forever frame transport, I don't think we need to use escape sequences for these characters.

I'd be interested in what @GrabYourPitchforks thinks though. Maybe this would make it too easy for our costumers to put unsanitized strings into their HTML.

Reply:
Well costumers should be getting the deserialized content which is encoded on the server as utf8 and decoded on the client as utf8, which should be as safe as putting any other content from an external source into your HTML.

Reply:
> costumers

![A dog in a panda costume running towards the camera](https://www.bing.com/th?id=OGC.f28f7ba72fcc48e2ed196c237266686e&pid=1.7&rurl=https%3a%2f%2fmedia.giphy.com%2fmedia%2f26wkQBBppNQ1ih584%2fgiphy.gif&ehk=h3P7%2fDNTpXK%2fIEQw2gLeOg)